### PR TITLE
the layout is created

### PR DESCRIPTION
**Pull request by unidrelyn**
## Summary
the layout of the Login is created 

##Screen Shot
image

##Card Trello 
https://trello.com/c/xks0CNVn/9-routing-i-login

### CODE REVIEW

File: React/game/.babelrc [hunk 1]
```diff
@@ -9,8 +9,10 @@
         "~components": "./src/app/components",
         "~screen": "./src/app/screen",
         "~redux": "./src/redux",
-        "~square": "./src/app/screen/Game/components/Square",
-        "~board": "./src/app/screen/Game/components/Board"
+        "~square": "./src/app/screens/Game/components/Square",
+        "~board": "./src/app/screens/Game/components/Board",
+        "@login": "./src/app/screens/Login/components/Login",
+        "@redux": "./src/redux"
```
Comment (DLotero) [inline React/game/.babelrc]: why do we have both `~` and `@`?

File: React/game/src/app/screens/Login/components/Login/layout.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import { Field, reduxForm } from 'redux-form';
+
+import styles from './styles.scss';
+
+function LoginLayout(props) {
+  const { handleSubmit } = props;
+  return (
+    <form onSubmit={handleSubmit}>
+      <div className={styles.login}>
+        <div>
```
Comment (DLotero) [inline React/game/src/app/screens/Login/components/Login/layout.js]: are we using this div for anything?

Comment (DLotero) [reply React/game/src/app/screens/Login/components/Login/layout.js]: Also, remember to use PropTypes in every component

File: React/game/src/redux/auth/actions.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  AUTH_LOGIN: 'AUTH_LOGIN'
+};
+
+const actionCreators = {
+  submit: state => ({
+    type: actionsTypes.AUTH_LOGIN,
+    payload: state
+  }),
+  onLogin: () => dispatch => {
+    console.log('onLogin');
```
Comment (DLotero) [inline React/game/src/redux/auth/actions.js]: remove this console.log

File: React/game/src/redux/store.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+const composeEnhancers =
+  typeof window === 'object' && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
+    ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({})
+    : compose;
+/* eslint-enable */
+
+const enhancer = composeEnhancers(applyMiddleware(thunk));
+const store = createStore(combineReducers(reducers), enhancer);
+
+export default store;
```
Comment (DLotero) [inline React/game/src/redux/store.js:22]: add eof line

File: React/game/src/redux/auth/reducer.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  switch (action.type) {
+    case actionsTypes.AUTH_LOGIN:
+      return {
+        ...state,
+        hasError: action.payload
+      };
+    default:
+      return state;
+  }
+}
```
Comment (DLotero) [inline React/game/src/redux/auth/reducer.js:17]: add eof line

File: React/game/src/redux/auth/actions.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+    dispatch({
+      type: actionsTypes.AUTH_LOGIN,
+      payload: true
+    });
+  }
+};
+
+export default actionCreators;
```
Comment (DLotero) [inline React/game/src/redux/auth/actions.js:18]: add eof line

File: React/game/src/index.js [hunk 7]
```diff
@@ -1,10 +1,16 @@
...
+import registerServiceWorker from './registerServiceWorker';
 
-ReactDOM.render(<App />, document.getElementById('root'));
+ReactDOM.render(
+  <Provider store={store}>
+    <App />
+  </Provider>,
+  document.getElementById('root')
+);
 registerServiceWorker();
```
Comment (DLotero) [inline React/game/src/index.js:15]: add eof line

File: React/game/src/config/api.js [hunk 8]
```diff
@@ -0,0 +1,8 @@
+import { create } from 'apisauce';
+
+const api = create({
+  baseURL: process.env.API_BASE_URL,
+  timeout: 5000
+});
+
+export default api;
```
Comment (DLotero) [inline React/game/src/config/api.js:8]: add eof line

File: React/game/src/app/screens/Login/components/Login/styles.scss [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+.button {
+  color: $blue;
+  @extend %login;
+  text-shadow: 1px 1px $grey;
+  width: 50%;
+  &:active {
+    background: $green;
+    border: 2px solid $red;
+  }
+}
```
Comment (DLotero) [inline React/game/src/app/screens/Login/components/Login/styles.scss:43]: add eof line

File: React/game/src/app/screens/Login/components/Login/layout.js [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+        </div>
+        <button className={styles.button} type="submit">
+          Submit
+        </button>
+      </div>
+    </form>
+  );
+}
+
+export default reduxForm({ form: 'loginForm' })(LoginLayout);
```
Comment (DLotero) [inline React/game/src/app/screens/Login/components/Login/layout.js]: add eof line

File: React/game/src/services/BookService.js [hunk 11]
```diff
@@ -0,0 +1,5 @@
+import api from '../../config/api';
+
+export default {
+  getBookDetail: id => api.get('/books', { id })
+};
```
Comment (DLotero) [inline React/game/src/services/BookService.js]: add eof line

**Review by DLotero — CHANGES_REQUESTED** [review]

File: React/game/.babelrc (hunk 1 above)
Comment (unidrelyn) [reply React/game/.babelrc]: I'm migrating to change all '@'

Comment (unidrelyn) [reply React/game/.babelrc]: 👍

File: React/game/src/app/screens/Login/components/Login/layout.js (hunk 2 above)
Comment (unidrelyn) [reply React/game/src/app/screens/Login/components/Login/layout.js]: 👍

File: React/game/src/redux/auth/actions.js (hunk 3 above)
Comment (unidrelyn) [reply React/game/src/redux/auth/actions.js]: 👍

File: React/game/src/redux/store.js (hunk 4 above)
Comment (unidrelyn) [reply React/game/src/redux/store.js:22]: 👍

File: React/game/src/redux/auth/reducer.js (hunk 5 above)
Comment (unidrelyn) [reply React/game/src/redux/auth/reducer.js:17]: 👍

File: React/game/src/redux/auth/actions.js (hunk 6 above)
Comment (unidrelyn) [reply React/game/src/redux/auth/actions.js:18]: 👍

File: React/game/src/config/api.js (hunk 8 above)
Comment (unidrelyn) [reply React/game/src/config/api.js:8]: 👍

File: React/game/src/app/screens/Login/components/Login/styles.scss (hunk 9 above)
Comment (unidrelyn) [reply React/game/src/app/screens/Login/components/Login/styles.scss:43]: 👍

File: React/game/src/app/screens/Login/components/Login/layout.js (hunk 10 above)
Comment (unidrelyn) [reply React/game/src/app/screens/Login/components/Login/layout.js]: 👍

File: React/game/src/services/BookService.js (hunk 11 above)
Comment (unidrelyn) [reply React/game/src/services/BookService.js]: 👍

File: React/game/src/index.js [hunk 12]
```diff
@@ -1,10 +1,16 @@
 import React from 'react';
 import ReactDOM from 'react-dom';
+import { Provider } from 'react-redux';
+import store from '@redux/store';
+import App from '@components/App';
+// import { BrowserRouter as Router, Route } from 'react-router-dom';
```
Comment (DLotero) [inline React/game/src/index.js]: please delete this line

File: React/game/src/services/BookService.js (hunk 11 above)
Comment (DLotero) [inline React/game/src/services/BookService.js]: why do we have a `BookService` file? It this isn't used anywhere, please delete it

**Review by DLotero — CHANGES_REQUESTED** [review]

File: React/game/src/app/components/App/index.js [hunk 13]
```diff
@@ -1,7 +1,33 @@
...
+import Game from '@game';
 
-const App = () => <Game />;
+import Login from '@login';
 
-export default App;
+class App extends Component {
+  render() {
+    const isLogin = this.props.isLogin;
+    const validateRoute = () => (isLogin ? <Redirect to="/game" /> : <Login />);
```
Comment (DLotero) [inline React/game/src/app/components/App/index.js:12]: I think this works for now, but something tells me this code will be problematic once we have more than a `login` screen and a `game` screen

**Review by DLotero — CHANGES_REQUESTED** [review]
some small changes and comments

File: React/game/src/index.js (hunk 12 above)
Comment (unidrelyn) [reply React/game/src/index.js]: 👍

File: React/game/src/services/BookService.js (hunk 11 above)
Comment (unidrelyn) [reply React/game/src/services/BookService.js]: was created by an example of the card, I'm going to erase

Comment (unidrelyn) [reply React/game/src/services/BookService.js]: 👍

File: React/game/src/app/components/App/index.js (hunk 13 above)
Comment (unidrelyn) [reply React/game/src/app/components/App/index.js:12]: :(

File: React/game/.babelrc [hunk 14]
```diff
@@ -6,11 +6,14 @@
...
-        "~screen": "./src/app/screen",
-        "~redux": "./src/redux",
-        "~square": "./src/app/screen/Game/components/Square",
-        "~board": "./src/app/screen/Game/components/Board"
+        "@components": "./src/app/components",
+        "@screen": "./src/app/screen",
+        "@redux": "./src/redux",
+        "@square": "./src/app/screens/Game/components/Square",
+        "@board": "./src/app/screens/Game/components/Board",
+        "@login": "./src/app/screens/Login/components/Login",
```
Comment (wfolini) [inline React/game/.babelrc]: I'd delete square, board and login aliases, it's a bad practice put alias per screen.

**Review by DLotero — DISMISSED** [review]

File: React/game/src/app/components/App/index.js (hunk 13 above)
Comment (wfolini) [reply React/game/src/app/components/App/index.js:12]: Yep, this in next PRs need to change

File: React/game/src/app/screens/Login/components/Login/layout.js [hunk 15]
```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import { Field, reduxForm } from 'redux-form';
+
+import styles from './styles.scss';
+
+const validate = values => {
```
Comment (wfolini) [inline React/game/src/app/screens/Login/components/Login/layout.js]: extract this validation function to `src/utils/formValidations.js` file, in order to be reused in another forms

File: React/game/src/app/screens/Login/components/Login/layout.js [hunk 16]
```diff
@@ -0,0 +1,49 @@
...
+  }
+  if (!values.password) {
+    errors.password = 'Required';
+  } else if (values.password.length < 8) {
+    errors.password = 'invalid Password';
+  }
+  return errors;
+};
+
+const renderField = ({ input, label, type, meta: { touched, error, warning } }) => (
```
Comment (wfolini) [inline React/game/src/app/screens/Login/components/Login/layout.js]: Extract this component in another file. Remember, one component per file.

File: React/game/src/app/screens/Login/components/Login/layout.js [hunk 17]
```diff
@@ -0,0 +1,49 @@
...
+        <Field name="password" component={renderField} type="password" placeholder="*****" />
+        <button className={styles.button} type="submit">
+          Submit
+        </button>
+      </div>
+    </form>
+  );
+}
+
+export default reduxForm({ form: 'loginForm', validate })(LoginLayout);
```
Comment (wfolini) [inline React/game/src/app/screens/Login/components/Login/layout.js:25]: Extract `loginForm` as a const in `src/constants/formNames.js` and use here. Think about If you want to refer to a form, you need the name. And for some reason u change this name, u need to change it in every places where u use it. So, is more easy to do that extracting it as a const in separated file.

File: React/game/src/index.js (hunk 12 above)
Comment (wfolini) [reply React/game/src/index.js]: Don't forget to delete this

**Review by wfolini — CHANGES_REQUESTED** [review]
Excellent! some changes and go!

File: React/game/.babelrc (hunk 14 above)
Comment (unidrelyn) [reply React/game/.babelrc]: 👍

File: React/game/src/app/screens/Login/components/Login/layout.js (hunk 15 above)
Comment (unidrelyn) [reply React/game/src/app/screens/Login/components/Login/layout.js]: 👍

File: React/game/src/app/screens/Login/components/Login/layout.js (hunk 16 above)
Comment (unidrelyn) [reply React/game/src/app/screens/Login/components/Login/layout.js]: 👍

File: React/game/src/app/screens/Login/components/Login/layout.js (hunk 17 above)
Comment (unidrelyn) [reply React/game/src/app/screens/Login/components/Login/layout.js:25]: 👍

**Review by wfolini — APPROVED** [review]
👌

**Pull request merged**